apis/python/test/common.py

```diff
@@ -7,6 +7,14 @@
 import tiledb
 from tiledb.vector_search.storage_formats import storage_formats, STORAGE_VERSION
 
+def get_test_data_path():
+    current_dir = os.path.dirname(__file__)
+    test_data_path = os.path.join(current_dir, "../../../test-data")
```

Comment:
We need to have an absolute path for this.  The Clion debugger, for instance, will run the tests from a different directory.  One can get the top-level python directory (similar to CMAKE_SOURCE_DIR).

Reply:
Mmm I'm a bit confused... specific questions:
1. Are you using Clion to run the Python tests? If so, could you share with me your setup?
2. Are you saying that CLion will copy the `test_foo.py` and `common.py` files into a different temporary directory and then run them? And that's why relative path won't work?
3. I don't see `array_defs.py` in #173, I just see `array_defs.h`? Could you send me a link to `array_defs.py`?

<img width="450" alt="Screenshot 2024-01-16 at 6 14 59 PM" src="https://github.com/TileDB-Inc/TileDB-Vector-Search/assets/1396242/5de4ea3f-d7df-4274-81bb-02db54c7633e">

Reply:
Not 100% sure of how Clion does this, tbh — but, yes, I think it copies.  When running things under the debugger there is definitely an original and a copied version of the .py files. Also, CLion doesn’t run pytest directly, when invoking unit tests.  I did some poking around in the toolchain but didn’t see anything obvious.  Sufficiently advanced technology, unfortunately.

Mea culpa, I typed too fast — in Python I called the fille array_paths.py.  We should keep the same name between Python and C++ — I don’t have a strong preference between array_defs and array_paths.

I believe I attached array_paths.py to my earlier comments (had to rename it array_paths.txt) to my comments (github helpfully turns it into a hyperlink).  I'll send the .py file in slack.

Best Regards,
Andrew Lumsdaine

Reply:
Hmm I tested in CLion (running in both `Run` and `Debug` mode) and the relative paths look okay to me:

<img width="2032" alt="Screenshot 2024-01-16 at 9 40 42 PM" src="https://github.com/TileDB-Inc/TileDB-Vector-Search/assets/1396242/b6bc2456-9d2e-43e0-800f-2cd1fcec5cd5">

<img width="1555" alt="Screenshot 2024-01-16 at 9 13 38 PM" src="https://github.com/TileDB-Inc/TileDB-Vector-Search/assets/1396242/e3c0838c-695e-4aaa-92b6-2df16ed2e61d">

Is there some other configuration I should be using to see this issue?